tempesta_fw/http_parser.c

```diff
@@ -2518,12 +2518,15 @@ __parse_http_date(TfwHttpMsg *hm, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(I_DateDay) {
+		size_t n;
 		__fsm_sz = __data_remain(p);
 		/* Parse a 2-digit day. */
-		__fsm_n = parse_int_ws(p, __fsm_sz, &parser->_acc);
+		n = __fsm_sz < 3 ? __fsm_sz: 3;
+		__fsm_n = parse_int_ws(p, n, &parser->_acc);
 		if (__fsm_n == CSTR_POSTPONE) {
-			parser->_i_st = &&I_DateDay;
-			__msg_hdr_chunk_fixup(data, len);
+			if (n == 3)
```

Comment:
If we are parsing one byte at a time, `n` will always be `1`, so this check won't catch long numbers. So it will sometimes catch longer numbers, sometimes not. It's not like there will be a security breach, but something like `0016` would be possible if chunk border is between `00` and `16`.

I think it will be easier to make two explicit states, for the first and for second digits. Resulting code may end up even smaller. Day of month may be represented with a single digit, but that should not be an issue.

Reply:
Addressed.